octopusdeploy_framework/framework_provider.go

```diff
@@ -39,35 +40,46 @@ func (p *octopusDeployFrameworkProvider) Configure(ctx context.Context, req prov
 
 	config := Config{}
 	config.ApiKey = providerData.ApiKey.ValueString()
+	if config.ApiKey == "" {
+		config.ApiKey = os.Getenv("OCTOPUS_APIKEY")
+	}
 	config.Address = providerData.Address.ValueString()
+	if config.Address == "" {
+		config.Address = os.Getenv("OCTOPUS_URL")
+	}
 	config.SpaceID = providerData.SpaceID.ValueString()
 	if err := config.GetClient(ctx); err != nil {
 		resp.Diagnostics.AddError("failed to load client", err.Error())
 	}
+	if err := config.GetClient(ctx); err != nil {
+		resp.Diagnostics.AddError("failed to load client", err.Error())
+	}
 
 	resp.DataSourceData = &config
 	resp.ResourceData = &config
 }
 
 func (p *octopusDeployFrameworkProvider) DataSources(ctx context.Context) []func() datasource.DataSource {
-	return []func() datasource.DataSource{}
+	return []func() datasource.DataSource{
+		NewProjectGroupsDataSource,
+	}
 }
 
 func (p *octopusDeployFrameworkProvider) Resources(ctx context.Context) []func() resource.Resource {
-	return []func() resource.Resource{}
+	return []func() resource.Resource{
+		NewProjectGroupResource,
+	}
 }
 
 func (p *octopusDeployFrameworkProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"address": schema.StringAttribute{
-				Optional:    false,
```

Comment:
These properties were throwing errors on tests. Comparing to the old provider `address` and `api_key` were required, however this supported default funcs. So the value could come from environment variables or the configuration. In the plugin framework these default funcs do not exist, so we set the values in the configure method